external-crates/move/crates/move-compiler/src/naming/translate.rs

```diff
@@ -501,45 +567,131 @@ impl<'env> Context<'env> {
             .nominal_blocks
             .iter()
             .rev()
-            .find(|(block_name, _, _)| name == *block_name)
-            .map(|(_, id, block_type)| (id, block_type));
+            .find(|(_, block_name, _, _)| block_name.is_some_and(|n| n == name))
+            .map(|(_, _, id, block_type)| (id, block_type));
         if let Some((id, block_type)) = id_opt {
-            if *block_type == expected_block_type {
-                let nvar_ = N::Var_ {
-                    name,
-                    id: *id,
-                    color: 0,
-                };
-                Some(BlockLabel(sp(loc, nvar_)))
+            let block_type = *block_type;
+            if block_type.is_acceptable_usage(usage) {
+                Some(block_label(
+                    loc,
+                    NominalBlockUsage::Return,
+                    Some(name),
+                    *id,
+                    block_type,
+                ))
```

Comment:
Git won't let me do this plus the code above, but... same changes:

```suggestion
            let block_type = *block_type;
            if block_type.is_acceptable_usage(usage) {
                Some(label)
```